sdk/storage/Azure.Storage.Queues/src/QueueMessageCodec.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+
+namespace Azure.Storage.Queues
+{
+    internal class QueueMessageCodec
+    {
+        public static string EncodeMessageBody(BinaryData binaryData, QueueMessageEncoding messageEncoding)
+        {
+            if (binaryData == null)
+            {
+                return null;
+            }
+            switch (messageEncoding)
+            {
+                case QueueMessageEncoding.UTF8:
+                    return binaryData.ToString();
+                case QueueMessageEncoding.Base64:
+                    return Convert.ToBase64String(binaryData.ToArray());
```

Comment:
binaryData.ToMemory would avoid copying to a separate array.

Reply:
Unfortunately https://docs.microsoft.com/en-us/dotnet/api/system.convert.tobase64string?view=netcore-3.1 doesn't have overload that would take readonlymemory or span...

Reply:
You can use MemoryMarshal.TryGetArray - https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.memorymarshal.trygetarray?view=netcore-3.1

Reply:
sounds good. it's getting complicated though.